shoop/themes/classic_gray/views/product_preview.py

```diff
@@ -8,7 +8,21 @@
 from shoop.front.views.product import ProductDetailView
 
 
+class CGProductDetailView(ProductDetailView):
+    def get_context_data(self, **kwargs):
+        # By default the template rendering the basket add form
+        # uses the `request.path` as its' `next` value.
+        # This is fine if you are on product page but here in
+        # preview, we cannot redirect back to `/xtheme/product_preview`.
+
+        context = super(CGProductDetailView, self).get_context_data(**kwargs)
+        # Add `return_url` to context to avoid usage of `request.path` in
```

Comment:
I still don't understand _why_ this was added.

The return URL is used when AJAX-based basket updating is disabled; instead of returning to the product page when a product has been added to the basket, we return here? Why?